### DISCUSSION TITLE
AutoApprove a PR only after all Status Checks pass successfully

### DISCUSSION THREAD

**Discussion by gioce90** [renovatebot/renovate#32947] [Suggest an Idea]
### Tell us more.

With issue #9685 and #21576, we have the `autoApprove` setting.
It would be great if this approval could be conditioned to occur only when all the Status checks of the PR have passed successfully.

Here’s an example: in my workflow with Azure DevOps, PRs can only be completed after a review from three developers. This setup works well for feature development or bug fixes created by other devs. However, for Renovate PRs, one or two reviewers are usually sufficient.
The autoApprove setting is especially useful in this scenarios with a "minimal number of reviewers" policy, as the Renovate bot itself can approve the PR. 
However, it feels odd to see an approval when some status checks, such as builds or unit tests, have failed.

Would it be possible to delay this automatic approval until all status checks for the PR have passed successfully?

**Comment by rarkins** [comment]
Possible but impractical. You'd have to wait until the next time Renovate runs, so then your developers would be wondering why it's not auto approved. Why do feelings matter here? It should only merge if it is approved and passes. An approved but failing PR shouldn't be any threat.

**Reply by gioce90** [reply]
> You'd have to wait until the next time Renovate runs

I’ve thought about it too, and I believe that perhaps, for Azure DevOps, a Status Check Policy that verifies all other PR policies have been successfully resolved before applying the Approve (I’m not sure if this is technically possible yet) could solve the issue. I’ll investigate.

As I’ve already mentioned, it just feels a bit strange to see a PR Approval — even if it’s from a bot — when builds or tests are failing.
